### PR TITLE
Simplify and expand logging in is_resolving_htlc_output

### PR DESCRIPTION
**Pull request by TheBlueMatt**
Based on #282 since they conflict.

### CODE REVIEW

File: src/ln/channelmonitor.rs [hunk 1]
```diff
@@ -1860,31 +1881,28 @@ impl ChannelMonitor {
...
+						"our previous local commitment tx", true);
 				}
 			}
 			if let Some(&(ref htlc_outputs, ref htlc_sources)) = self.remote_claimable_outpoints.get(&input.previous_output.txid) {
-				scan_commitment!(htlc_outputs, htlc_sources, "remote commitment tx");
+				scan_commitment!(htlc_outputs, htlc_sources, "remote commitment tx", false);
 			}
 
 			// If tx isn't solving htlc output from local/remote commitment tx and htlc isn't outbound we don't need
 			// to broadcast solving backward
```
Comment (ariard) [inline src/ln/channelmonitor.rs]: nit: Maybe change the term "broadcast" to "relay" or other, I mean we should enforce terminology to dissociate "broadcasting tx onchain" from to "relay state offchain" ?

File: src/ln/channelmonitor.rs [hunk 2]
```diff
@@ -1860,31 +1881,28 @@ impl ChannelMonitor {
...
-						"our previous local commitment tx");
+						"our previous local commitment tx", true);
 				}
 			}
 			if let Some(&(ref htlc_outputs, ref htlc_sources)) = self.remote_claimable_outpoints.get(&input.previous_output.txid) {
-				scan_commitment!(htlc_outputs, htlc_sources, "remote commitment tx");
+				scan_commitment!(htlc_outputs, htlc_sources, "remote commitment tx", false);
 			}
 
 			// If tx isn't solving htlc output from local/remote commitment tx and htlc isn't outbound we don't need
```
Comment (ariard) [inline src/ln/channelmonitor.rs]: nit: Oh and comment could be more meaningful, "filter for tx solving HTLC Offered Output on Local Commitment tx via offered_preimage_claim or HTLC Received Output on Remote Commitment tx via accepted_preimage_claim"

File: src/ln/channelmonitor.rs [hunk 3]
```diff
@@ -1833,22 +1833,43 @@ impl ChannelMonitor {
...
+			let offered_preimage_claim = input.witness.len() == 3 && input.witness[2].len() == OFFERED_HTLC_SCRIPT_WEIGHT;
+
+			macro_rules! log_claim {
+				($source: expr, $local_tx: expr, $outbound_htlc: expr, $payment_hash: expr, $source_avail: expr) => {
+					// We found the output in question, but aren't failing it backwards
+					// as we have no corresponding source. This implies either it is an
+					// inbound HTLC or an outbound HTLC on a revoked transaction.
+					if ($local_tx && revocation_sig_claim) ||
+							($outbound_htlc && !$source_avail && (accepted_preimage_claim || offered_preimage_claim)) {
+						log_error!(self, "Input spending {}:{} in {} resolves {} HTLC with payment hash {} from {} with {}!",
```
Comment (ariard) [inline src/ln/channelmonitor.rs]: nit: place tx label near its identifiers : "Input spending ($source) previous_output.txid:vout in ..."

Comment (TheBlueMatt) [reply src/ln/channelmonitor.rs]: Good idea, done.

File: src/ln/channelmonitor.rs (hunk 2 above)
Comment (TheBlueMatt) [reply src/ln/channelmonitor.rs]: Updated the comment a bit and mostly pointed to a new comment in scan_commitment.

File: src/ln/channelmonitor.rs (hunk 1 above)
Comment (TheBlueMatt) [reply src/ln/channelmonitor.rs]: Done.

**Pull request merged**